tfx/components/example_gen/utils.py

```diff
@@ -101,36 +140,15 @@ def dict_to_example(instance: Dict[str, Any]) -> example_pb2.Example:
     if isinstance(pyval, bytes):
       pyval = pyval.decode(_DEFAULT_ENCODING)
 
-    if pyval is None:
-      feature[key] = feature_pb2.Feature()
-    elif isinstance(pyval, int):
-      feature[key] = feature_pb2.Feature(
-          int64_list=feature_pb2.Int64List(value=[pyval]))
-    elif isinstance(pyval, float):
-      feature[key] = feature_pb2.Feature(
-          float_list=feature_pb2.FloatList(value=[pyval]))
-    elif isinstance(pyval, str):
-      feature[key] = feature_pb2.Feature(
-          bytes_list=feature_pb2.BytesList(
-              value=[pyval.encode(_DEFAULT_ENCODING)]))
+    if isinstance(pyval, _SINGLE_VALUE_TYPES):
+      feature[key] = value_to_feature(pyval)
+    
     elif isinstance(pyval, list):
-      if not pyval:
-        feature[key] = feature_pb2.Feature()
-      elif isinstance(pyval[0], int):
-        feature[key] = feature_pb2.Feature(
-            int64_list=feature_pb2.Int64List(value=pyval))
-      elif isinstance(pyval[0], float):
-        feature[key] = feature_pb2.Feature(
-            float_list=feature_pb2.FloatList(value=pyval))
-      elif isinstance(pyval[0], str):
-        feature[key] = feature_pb2.Feature(
-            bytes_list=feature_pb2.BytesList(
-                value=[v.encode(_DEFAULT_ENCODING) for v in pyval]))
-      else:
-        raise RuntimeError('Column type `list of {}` is not supported.'.format(
-            type(value[0])))
+      feature[key] = list_to_feature(pyval)
+
     else:
-      raise RuntimeError('Column type {} is not supported.'.format(type(value)))
+      raise RuntimeError(f'Key {key} with Column type {type(value)} is not supported.')
```

Comment:
Can you make the error message in consistent format as the other two.

Reply:
done